DirectX11/HackerContext.cpp

```diff
@@ -180,56 +174,37 @@ static ResourceSnapshot SnapshotResource(ID3D11Resource *handle)
 	return ResourceSnapshot(handle, hash, orig_hash);
 }
 
-void HackerContext::_RecordShaderResourceUsage(ShaderInfoData *shader_info, ID3D11ShaderResourceView *views[D3D11_COMMONSHADER_INPUT_RESOURCE_SLOT_COUNT])
+void HackerContext::_RecordShaderResourceUsage(ShaderInfoData& shader_info, ID3D11ShaderResourceView *views[D3D11_COMMONSHADER_INPUT_RESOURCE_SLOT_COUNT])
```

Comment:
This is probably ok, though I'd need to double check if it's safe to use a reference here - certainly if you don't know you need to change this, you probably shouldn't.

That aside, there is another slight issue with the coding style here, and that's the placement of the ampersand next to the type name rather than the variable name. In this particular context it doesn't matter too much, but I do have a preference on putting the * or & next to the type name everwhere other than return types, because of this ambiguity:

```
int* a, b, c;
```

Did I just declare three pointers, or a pointer and two ints?

Based on the placement of the * it looks like I declared three pointers, but I actually declared one pointer and two ints. Declaring it as such:
```
int *a, b, c;
```

Is slightly clearer as to how the language interprets it. You can argue that this is a very stupid part of C/C++ (and you would be absolutely right) and insist that the * is part of the type so it should be on the left, but it is what it is and I've seen this bite programmers before hence my preference to put it on the right with the variable name.


As mentioned elsewhere we haven't adopted a coding style for 3DMigoto, so take this as a suggestion with justification rather than a rule. I won't block new code that goes against this suggestion if you do prefer it the other way, but I would prefer that existing code not be altered.